tests/unit/package_managers/test_pip.py

```diff
@@ -3771,6 +3840,7 @@ def convert_to_component_list(packages: list[dict]) -> list[Component]:
 
         for package in packages:
             for dependency in package["dependencies"]:
+                dependency["purl"] = pip._generate_purl_dependency(dependency)
```

Comment:
Can we have the purls as literal strings in the `expect_package_*` data instead of generating them dynamically?